cub/cub/util_namespace.cuh

```diff
@@ -177,14 +129,17 @@
 #  define CUB_DETAIL_MAGIC_NS_END
 #else // not defined(CUB_DISABLE_NAMESPACE_MAGIC)
 #  if defined(_NVHPC_CUDA)
-#    define CUB_DETAIL_MAGIC_NS_BEGIN inline namespace CUB_DETAIL_MAGIC_NS_NAME(CUB_VERSION, NV_TARGET_SM_INTEGER_LIST) {
+#    define CUB_DETAIL_MAGIC_NS_BEGIN                                                                               \
+      inline namespace THRUST_DETAIL_SPLICE_WITH_UNDERSCORE(CUB, CUB_VERSION, SM, NV_TARGET_SM_INTEGER_LIST, NVHPC) \
+      {
 #    define CUB_DETAIL_MAGIC_NS_END }
 #  else // not defined(_NVHPC_CUDA)
-#    define CUB_DETAIL_MAGIC_NS_BEGIN inline namespace CUB_DETAIL_MAGIC_NS_NAME(CUB_VERSION, __CUDA_ARCH_LIST__) {
+#    define CUB_DETAIL_MAGIC_NS_BEGIN                                                                 \
+      inline namespace THRUST_DETAIL_SPLICE_WITH_UNDERSCORE(CUB, CUB_VERSION, SM, __CUDA_ARCH_LIST__) \
```

Comment:
I considered adding a different tag if the compiler is not NVHPC, but then I would need to distinguish clang in CUDA mode and nvcc, which we don't do and need today. But I am open to add it.